Migration to DNX RC1

alyiox opened the pull request.

hekaiduo committed:
Add IAsyncQueryExecutor and IKeySerializer

hekaiduo committed:
Add configurable expiration span

hekaiduo committed:
update dependencies (i.e. Json.Net)

hekaiduo committed:
change version number

hekaiduo committed:
edit nuspec files

hekaiduo committed:
fix a bug for MemoryCacheBasedQueryCache (cause: msdn doc is wrong/misleading for AddOrGetExisting(CacheItem,CachePolicy)

hekaiduo committed:
update assembly version

hekaiduo committed:
Add EmptyQueryCache

hekaiduo committed:
Add test for EmptyQueryCache()

hekaiduo committed:
update metadata

alyiox committed:
Migrate to DNX RC1

karldodd commented on src/DotQuery.Core/Queries/AggregateAsyncQueryClientSideExecutor.cs:
```diff
@@ -1,94 +1,94 @@
+                while (taskList.Count > 0)
+                {
+#if portable
+                    Task<TResult> completedTask = await TaskEx.WhenAny(taskList);                    
```
I guess the adoption of `TaskEx` can be removed this time.

alyiox replied:
fixed

karldodd commented on src/DotQuery.Core/project.json:
```diff
@@ -0,0 +1,26 @@
+  "copyright": "Copyright ©  2013",
+
+  "dependencies": {
+    "Newtonsoft.Json": "6.0.3"
```
Let's upgrade to Json.Net 7.0.1

alyiox replied:
fixed

karldodd commented on src/DotQuery.Extensions/project.json:
```diff
@@ -0,0 +1,33 @@
+  },
+  "dependencies": {
+    "DotQuery.Core": "1.2.0-beta1",
+    "Newtonsoft.Json": "6.0.3",
```
Upgrade to Json.Net 7.0.1

alyiox replied:
fixed

karldodd commented on src/DotQuery.Extensions/project.json:
```diff
@@ -0,0 +1,33 @@
+  "dependencies": {
+    "DotQuery.Core": "1.2.0-beta1",
+    "Newtonsoft.Json": "6.0.3",
+    "System.Collections.Concurrent": "4.0.11-beta-23516"
```
Why is `System.Collections.Concurrent` here rather than in frameworks->dependencies?

alyiox replied:
fixed,
btw, it was generated by the vs 2015 automatically.

alyiox committed:
Refactor codebase

- Update DNX 1.0.0-rc1-update2 to 1.0.0-rc1-final

- Remove unnecessary usings

- Update Newtonsoft.Json from 6.0.3 to 7.0.1

alyiox commented:
Hi,
the code base has been refactored, please review it.
